### PR TITLE
[DDW-809] Analytics - part 2 - Matomo client, router-level tracking and new Terms of Service

### PR DESCRIPTION
**Pull request by marcin-mazurek**
This PR introduces a Matomo client, implements router-level tracking with Matomo, example of component-level tracking for stake pool view type change, as well as updates the Terms of Service.

This PR should not be tested by QA team as it does not include full tracking - which is delivered in the third PR.

First part: https://github.com/input-output-hk/daedalus/pull/2927
Third part: https://github.com/input-output-hk/daedalus/pull/3003

## Screenshots

No UI changes except for TOS content.

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1654839266665729)
- [ ] Test

---

### Basics

- [x] PR assigned to the PR author(s)
- [x] `input-output-hk/daedalus-dev` and `input-output-hk/daedalus-qa` assigned as PR reviewers
- [x] If there are UI changes, Alexander Rukin assigned as an additional reviewer
- [x] All visual regression testing has been reviewed (assign `run Chromatic` label to PR to trigger the run)
- [x] PR has appropriate labels (`release-vNext`, `feature`/`bug`/`chore`, `WIP`)
- [x] PR link is added to a Jira ticket, ticket moved to In Review
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [x] PR has a good description that summarizes all changes
- [x] PR contains screenshots (in case of UI changes)
- [x] CHANGELOG entry has been added to the top of the appropriate section (_Features_, _Fixes_, _Chores_) and is linked to the correct PR on GitHub
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [x] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [x] Japanese text changes are proofread and approved (Junko Oda)
- [x] Storybook works and no stories are broken (`yarn storybook`)
- [x] In case of dependency changes `yarn.lock` file is updated

### Code Quality

- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with typescript types
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing

- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review

- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/components/staking/stake-pools/StakePools.tsx [hunk 1]
```diff
@@ -119,29 +119,47 @@ class StakePools extends Component<Props, State> {
...
+
+  handleGridView = () => {
     this.setState({
       isGridView: true,
       isGridRewardsView: false,
       isListView: false,
     });
+
+    this.props.analyticsClient.sendEvent(
+      'Stake Pools',
```
Comment (lucas-barros) [inline source/renderer/app/components/staking/stake-pools/StakePools.tsx:136]: Should we make these event names as const so its better/safer to reuse?

File: source/main/ipc/electronStoreConversation.ts [hunk 2]
```diff
@@ -25,29 +25,7 @@ const unset = async (key: StorageKey) =>
...
-  await unset(keys.THEME);
-  await unset(keys.USER_DATE_FORMAT_ENGLISH);
-  await unset(keys.USER_DATE_FORMAT_JAPANESE);
-  await unset(keys.USER_LOCALE);
-  await unset(keys.USER_NUMBER_FORMAT);
-  await unset(keys.USER_TIME_FORMAT);
-  await unset(keys.WALLET_MIGRATION_STATUS);
-  await unset(keys.WALLETS);
-  await unset(keys.WINDOW_BOUNDS);
+  await Promise.all(Object.values(keys).map(unset));
```
Comment (shawnbusuttil) [inline source/main/ipc/electronStoreConversation.ts:28]: Well done.

File: source/renderer/app/analytics/MatomoClient.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    );
+  }
+
+  sendPageNavigationEvent = async (pageTitle: string) => {
+    this.matomoTracker.track({
+      _id: this.userId,
+      action_name: pageTitle,
+      url: this.getAnalyticsURL(),
+      [CPU_DIMENSION_KEY]: formatCpuInfo(this.environment.cpu),
+      [RAM_DIMENSION_KEY]: Math.round(
```
Comment (shawnbusuttil) [inline source/renderer/app/analytics/MatomoClient.ts:36]: I suggest extracting this and keeping it in utils.

File: source/renderer/app/components/staking/stake-pools/StakePools.tsx (hunk 1 above)
Comment (shawnbusuttil) [reply source/renderer/app/components/staking/stake-pools/StakePools.tsx:136]: I agree. Same for `pageTitle`.

File: source/renderer/app/analytics/MatomoClient.ts (hunk 3 above)
Comment (marcin-mazurek) [reply source/renderer/app/analytics/MatomoClient.ts:36]: Good suggestion, and actually we already have a utility to convert from bytes to human friendly value so I reused it.

File: source/renderer/app/components/staking/stake-pools/StakePools.tsx (hunk 1 above)
Comment (marcin-mazurek) [reply source/renderer/app/components/staking/stake-pools/StakePools.tsx:136]: Ok, pushing this change in https://github.com/input-output-hk/daedalus/pull/3003

Comment (marcin-mazurek) [reply source/renderer/app/components/staking/stake-pools/StakePools.tsx:136]: Got it. I think it's basically the same story as here: https://github.com/input-output-hk/daedalus/pull/3003#discussion_r913620638 - whatever we agree should be applied consistently.

File: package.json [hunk 4]
```diff
@@ -241,7 +241,8 @@
     "inquirer": "7.3.3",
     "json-bigint": "1.0.0",
     "lodash": "4.17.21",
-    "lodash-es": "4.17.21",
+    "lodash-es": "4.17.15",
+    "matomo-tracker": "^2.2.4",
```
Comment (lucas-barros) [inline package.json:245]: Should we set the this package to a fixed version?

Comment (marcin-mazurek) [reply package.json:245]: Good catch, fixed

**Review by lucas-barros — APPROVED** [review]

**Review by shawnbusuttil — APPROVED** [review]

**Pull request closed without merging**